### ISSUE TITLE
Higher level types for Values and DataArrays

### ISSUE THREAD

**Issue by gicmo** [G-Node/nix#665]
Currently we only support 'primitive' types, e.g. everything that is in the `DataType`. How should we handle more complex times like:
- DateTimes (needs storage type and interpretation, e.g. int as seconds since Epoch, string as ISO 8601)
- Complex Numbers
- Images (encoding, format, ...)

**Comment by msuzen** [comment]
Couple of ideas out side of boost family:

* DataTimes, using Rs easy conversions. [link](https://rdrr.io/rforge/RcppClassicExamples/man/RcppDate.html)

* More advanced mathematical objects,  [IT++](https://en.wikipedia.org/wiki/IT%2B%2B) like library would be useful.

* For images and video  OpenCV objects can be used out of box, i.e., [imread & co](http://docs.opencv.org/2.4/modules/highgui/doc/reading_and_writing_images_and_video.html?highlight=imread). I haven't tested but looks like OpenCV has hdf5 calls too. [here](http://docs.opencv.org/trunk/d3/d23/classcv_1_1hdf_1_1HDF5.html).

**Comment by achilleas-k** [comment]
Well, one thing I've been hoping for is to drop boost as a dependency completely. Not sure how I feel about adding more back in.

**Comment by msuzen** [comment]
@achilleas-k I agree on reducing dependencies to a bare minimum and boost being a heavy-weight.

**Comment by msuzen** [comment]
Regarding, storing and extracting data-time objects, it is a community-wide issue on hdf5, c.f. [here](http://stackoverflow.com/questions/23570632/store-datetimes-in-hdf5-with-h5py).

**Comment by gicmo** [comment]
@achilleas-k yeah, should be done without libboost dep.
